namui-cli/src/build/build_server/code_watcher.rs

```diff
@@ -29,30 +28,33 @@ impl CodeWatcher {
             watching_paths: HashSet::new(),
             manifest_path,
             watcher_receiver,
-            was_changed_while_flushing_events: true,
         }
     }
 
-    pub fn wait_for_change(&self) {
-        loop {
-            match self.watcher_receiver.recv() {
-                Ok(event) => {
-                    match event {
-                        DebouncedEvent::Create(_)
-                        | DebouncedEvent::Remove(_)
-                        | DebouncedEvent::Rename(_, _)
-                        | DebouncedEvent::Write(_) => {
-                            return;
-                        }
-                        _ => (),
-                    };
+    pub fn wait_for_change(&mut self) {
+        let was_changed_while_flushing_events = self.flush_events();
+        match was_changed_while_flushing_events {
+            true => (),
+            false => loop {
+                match self.watcher_receiver.recv() {
```

Comment:
Before start waiting, It flush event queue and if there are necessary events, It return `()` immediately

Reply:
I have a question. Why flush before waiting?

It still have a problem as I commented here https://github.com/NamseEnt/namseent/pull/71#discussion_r783639092